system/ee/legacy/libraries/Messages.php

```diff
@@ -1409,7 +1409,7 @@ public function do_member_search()
                         ->first();
 
                     if ($role) {
-                        $search_query[] = " member_id IN (" . implode($role->Members->pluck('member_id'), ',') . ")";
+                        $search_query[] = " member_id IN (" . implode(',', $role->Members->pluck('member_id')) . ")";
```

Comment:
Just to confirm.. this one is more of a bug that PHP 8.1 support?

Reply:
This **is** PHP 8.1 support fix. In older versions of PHP, `implode` could accept parameters in any order. PHP 8.1 is strict here